crates/services/upgradable-executor/src/executor.rs

```diff
@@ -814,6 +824,9 @@ where
     /// This is a slow operation, and the cached result should be used if possible,
     /// for instancy by calling `get_module` below.
     #[cfg(feature = "wasm-executor")]
+    // TODO: This methods may compile the WASM binary inside. It can take a lot of time,
+    //  blocking the runtime. Instead, we should make this function async and compile
+    //  the WASM in a separate thread.
```

Comment:
Did we create an issue for this? Seems worth tracking as this could cause opaque bugs in the future.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2755

Reply:
It's probably best to included it inline.